src/cmd/serve.rs

```diff
@@ -240,6 +240,39 @@ fn rebuild_done_handling(broadcaster: &Sender, res: Result<()>, reload_path: &st
     }
 }
 
+fn construct_url(base_url: &str, no_port_append: bool, interface_port: u16) -> String {
```

Comment:
With all the dependencies we're pulling, we don't pull the [url](https://crates.io/crates/url) ?

Reply:
I don't believe so—at least not as a first-party dependency.

Reply:
we do pull it in components/libs/Cargo.toml

Reply:
Oh, I see how that's structured now. Please advise on how you'd like me to proceed so we can get this patch merged. This works as is but I can rewrite it with the url crate if you prefer.

Reply:
Let's use the url crate if it makes it simpler

Reply:
Having looked a little closer, I actually don't know that it makes it any simpler. The current implementation works well and is easy to follow IMO. If a maintainer wants to adjust it before merging that's totally fine.

Reply:
I personally think that the `url` crate would be an improvement since it would also allow validating URLs when parsing them from configs, but that can be reserved for a separate PR.